### PR TITLE
Update num loc from all paths

### LINKED ISSUE #6402: LOC wrong: cosmetic issue
Just did  Matching Brackets in Go and in the community solutions it shows it being 98 LOC when it is more like 23 LOC.
info
info_1

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes https://github.com/exercism/exercism/issues/6402

### CODE REVIEW

File: app/commands/iteration/destroy.rb [hunk 1]
```diff
@@ -9,12 +9,11 @@ def call
...
       Solution::Unpublish.(solution) if solution.published_iteration_id == iteration.id
+      Solution::UpdateNumLoc.(solution)
     end
 
     private
     memoize
-    def solution
-      iteration.solution
-    end
+    def solution = iteration.solution
```
Comment (ErikSchierboom) [inline app/commands/iteration/destroy.rb:17]: I'll sneak these endless method refactorings in from time to time :)

File: app/commands/iteration/destroy.rb [hunk 2]
```diff
@@ -9,12 +9,11 @@ def call
 
       Solution::Uncomplete.(solution) unless solution.iterations.not_deleted.exists?
       Solution::Unpublish.(solution) if solution.published_iteration_id == iteration.id
+      Solution::UpdateNumLoc.(solution)
```
Comment (ErikSchierboom) [inline app/commands/iteration/destroy.rb:12]: This makes sure that when an iteration is destroyed, the num_loc of the solution is updated (could be a no-op).
The scenarios are:

1. The solution was the single published iteration: update num_loc to latest iteration's num_loc
2. The solution was the latest published iteration: update num_loc to latest published iteration's num_loc
3. The solution had one published iteration and this was not it: don't update num_loc
4. The solution was not published and this was the latest iteration: update num_loc to latest published iteration's num_loc
5. The solution was not published and this wasn't the latest iteration: don't update num_loc

File: app/commands/solution/publish.rb [hunk 3]
```diff
@@ -13,7 +13,6 @@ def call
         ActiveRecord::Base.transaction do
           solution.update(published_at: Time.current)
           Solution::PublishIteration.(solution, iteration_idx)
-          Solution::UpdateSnippet.(solution)
```
Comment (ErikSchierboom) [inline app/commands/solution/publish.rb:16]: I've removed this as `Solution::UpdateSnippet` is now also called from `Solution::PublishIteration` (as per https://github.com/exercism/website/pull/2712).

File: app/jobs/calculate_lines_of_code_job.rb [hunk 4]
```diff
@@ -27,14 +27,6 @@ def perform(iteration)
...
-    iteration.solution.update_column(:num_loc, num_loc) if should_update_solution?(iteration)
-  end
-
-  private
-  def should_update_solution?(iteration)
-    solution = iteration.solution
-    return true if solution.published_iteration_id == iteration.id
-
-    solution.published_iteration_id.nil? && solution.latest_iteration == iteration
+    Solution::UpdateNumLoc.(iteration.solution)
```
Comment (ErikSchierboom) [inline app/jobs/calculate_lines_of_code_job.rb:30]: Dry-ing up the logic

**Review by iHiD — APPROVED** [review]

**Pull request merged**